src/java.base/share/classes/java/lang/reflect/Constructor.java

```diff
@@ -90,10 +90,15 @@ ConstructorRepository getGenericInfo() {
         var genericInfo = this.genericInfo;
         // lazily initialize repository if necessary
         if (genericInfo == null) {
-            // create and cache generic info repository
-            genericInfo =
-                ConstructorRepository.make(getSignature(),
-                                           getFactory());
+            var root = this.root;
+            if (root != null) {
+                genericInfo = root.getGenericInfo();
+            } else {
+                // create and cache generic info repository
+                genericInfo =
+                        ConstructorRepository.make(getSignature(),
+                                getFactory());
```

Comment:
```suggestion
                genericInfo = ConstructorRepository.make(getSignature(), getFactory());
```

Nit: This fits well in one line.